### DISCUSSION TITLE
Is it possible to support markup extension in selector?

### DISCUSSION THREAD

**Discussion by rabbitism** [AvaloniaUI/Avalonia#9086] [Ideas]
As discussed in https://github.com/AvaloniaUI/Avalonia/pull/8793, there is no performance issue to use x:static to standardize part names. But it's useless for selector because selector doesn't support markup extension. So is it possible to add markup extension support for selector?  
Grammar:
`Selector="^:pointerover /template/ ContentPresenter#{x:Static Button.PART_ContentPresenter}"`

I checked several parts in source code, and I think it's possible
1. In `SelectorGrammar.Parse`, add new state and corresponding handling of `State.Name` followed by `{`. 
2. Add new `ISyntax` as `ReferenceNameSyntax`
3. In `AvaloniaXamlIlSelectorTransformer`, add transform handling of `ReferenceNameSyntax` with existing `TypeResolver`.

The only drawback is that it's impossible to test if two NameSyntax/ReferenceNameSyntax are actually equal.

**Comment by maxkatz6** [comment]
I feel like it could be too much for the selectors syntax to allow any kind of markup extensions. Also it doesn't match by schematics, as we don't have a "markup" here, it's only a selector - a predicate/filter. 
Currently MarkupExtension returns an object, which can be a binding or a color or another style instance. How selector should interpreter that?

I would split it into smaller ideas:
1. x:Static support build in selectors syntax. `Button:staticName(ns|Consts.ButtonName)` or something (don't treat this syntax as final!). 
2. Custom selectors support. Technically it's possible already, but only if you construct selectors in the code behind. Selectors parser doesn't support custom ones.
3. Actual markup extensions in property selectors? I would not, as same tasks can be achieved _way cleaner_ with `Classes.isActive="{Binding IsActive}"` syntax. 

The problem with `Button#:static(...)` is that `#:static` part is expected to be a single selector, while by CSS selectors standard it's not. `Button:staticName(ns|Consts.ButtonName)`on other hand seems to have a valid CSS selector syntax. 

Probably `Button:name(ns|Consts.ButtonName)` ?

**Reply by maxkatz6** [reply]
CC @AvaloniaUI/core
